java-checks/src/main/java/org/sonar/java/checks/spring/ModelAttributeNamingConventionForSpELCheck.java

```diff
@@ -0,0 +1,116 @@
+
+  private static final MethodMatchers ADD_ATTRIBUTE_MATCHER_WITH_TWO_PARAMS = MethodMatchers.create()
+    .ofTypes("org.springframework.ui.Model")
+    .names("addAttribute")
+    .addParametersMatcher("java.lang.String", "java.lang.Object")
+    .build();
+
+  private static final MethodMatchers ADD_ATTRIBUTE_MATCHER_WITH_ONE_PARAM = MethodMatchers.create()
+    .ofTypes("org.springframework.ui.Model")
+    .names("addAllAttributes")
+    .addParametersMatcher("java.util.Map")
+    .build();
+
+  private static final MethodMatchers MAP_OF = MethodMatchers.create()
+    .ofTypes("java.util.Map")
+    .names("of", "ofEntries", "entry")
+    .withAnyParameters()
+    .build();
+
+  @Override
+  protected MethodMatchers getMethodInvocationMatchers() {
+    return MethodMatchers.or(ADD_ATTRIBUTE_MATCHER_WITH_TWO_PARAMS, ADD_ATTRIBUTE_MATCHER_WITH_ONE_PARAM);
+  }
+
+  @Override
+  protected void onMethodInvocationFound(MethodInvocationTree mit) {
+    ExpressionTree argumentTree = mit.arguments().get(0);
+    if (argumentTree.is(Tree.Kind.STRING_LITERAL)) {
+      checkStringLiteralAndReport(argumentTree, argumentTree);
+    } else if (argumentTree.is(Tree.Kind.IDENTIFIER)) {
+      checkIdentifier((IdentifierTree) argumentTree);
+    } else if (argumentTree.is(Tree.Kind.MEMBER_SELECT)) {
+      checkMemberSelect((MemberSelectExpressionTree) argumentTree);
+    } else if (argumentTree.is(Tree.Kind.METHOD_INVOCATION)) {
+      checkMethodInvocation((MethodInvocationTree) argumentTree);
+    }
+  }
+
+  private boolean checkStringLiteralAndReport(ExpressionTree tree, ExpressionTree reportTree) {
+    if (tree.is(Tree.Kind.STRING_LITERAL)) {
+      LiteralTree literalTree = (LiteralTree) tree;
+      String literalValue = LiteralUtils.getAsStringValue(literalTree);
+      Matcher matcher = pattern.matcher(literalValue);
+      if (!matcher.matches()) {
+        reportIssue(reportTree,
+          "Attribute names must begin with a letter (a-z, A-Z), underscore (_), or dollar sign ($) and can be followed by letters, digits, underscores, or dollar signs.");
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private void checkIdentifier(IdentifierTree identifierTree) {
+    VariableTreeImpl declaration = (VariableTreeImpl) identifierTree.symbol().declaration();
+    if (declaration != null) {
+      checkStringLiteralAndReport(declaration.initializer(), identifierTree);
+    }
+  }
+
+  private void checkMemberSelect(MemberSelectExpressionTree memberSelectExpressionTree) {
+    checkIdentifier(memberSelectExpressionTree.identifier());
+  }
+
+  private void checkMethodInvocation(MethodInvocationTree methodInvocationTree) {
+    if (MAP_OF.matches(methodInvocationTree)) {
+      for (int i = 0; i < methodInvocationTree.arguments().size(); i += 2) {
+        ExpressionTree key = methodInvocationTree.arguments().get(i);
+        if (checkStringLiteralAndReport(key, methodInvocationTree)) {
```

Comment:
Thanks for the update! Should we go further and check for keys that are identifiers, as well? It looks to me like we could apply the same logic we use `onMethodInvocationFound` here to complete the loop. WDYT?